kotlin-android/app/src/main/java/com/mongodb/tasktracker/ProjectActivity.kt

```diff
@@ -0,0 +1,114 @@
+package com.mongodb.tasktracker
+
+import android.app.AlertDialog
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.EditText
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mongodb.tasktracker.model.Project
+import com.mongodb.tasktracker.model.ProjectAdapter
+import com.mongodb.tasktracker.model.User
+import io.realm.Realm
+import io.realm.RealmList
+import io.realm.kotlin.where
+import io.realm.mongodb.sync.SyncConfiguration
+import java.util.*
+
+class ProjectActivity : AppCompatActivity() {
+    private var user: io.realm.mongodb.User? = null
+    private lateinit var realm: Realm
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var adapter: ProjectAdapter
+    private lateinit var fab: FloatingActionButton
+
+    override fun onStart() {
+        super.onStart()
+        try {
+            user = taskApp.currentUser()
+        } catch (e: IllegalStateException) {
+            Log.w(TAG(), e)
+        }
+        if (user == null) {
+            // if no user is currently logged in, start the login activity so the user can authenticate
+            startActivity(Intent(this, LoginActivity::class.java))
+        } else {
+            // configure realm to use the current user and the partition corresponding to "My Project"
+            val config = SyncConfiguration.Builder(user!!, "user=${user!!.id}")
+                .waitForInitialRemoteData()
+                .build()
+
+            // save this configuration as the default for this entire app so other activities and threads can open their own realm instances
+            Realm.setDefaultConfiguration(config)
```

Comment:
Is this the recommended pattern? This configuration is just for the user realm. We also have several other project realms that may need to be opened. Does setting a default configuration actually make it more clear?

Reply:
Removed this. The old app set a default configuration because we only ever needed a single realm, but the new app requires so many realms there is never a "default" one.